### PR TITLE
Ports - Heather

### PR DESCRIPTION
**Pull request by piffer59**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The Model talks to external databases
Describe in your own words what the Controller is doing in Rails | It is pulling the data from the Model using different actions.
Describe in your own words what the View is doing in Rails | Displays the data that has been pulled by the Controller
Describe an edge-case controller test you wrote |Returning a not found message when a task is not found.
What is the purpose of using strong params? (i.e. the params method in the controller) | Helps to increase security, they  prevent accidentally allowing users to update model attributes.
How are Rails migrations related to Rails models? | migrations allow you to change your model
Describe one area of Rails that are still unclear on | Testing Controller methods is still confusing

### CODE REVIEW

**Comment by tildeee** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | x
Answered comprehension questions | x
Successfully handles: Index, Show | x
Index & Show tests pass | x
Successfully handles: New, Create | x
New & Create tests pass | x
Successfully handles: Edit, Update | x
Tests for Edit & Update test valid & invalid task ids | x
Successfully handles: Destroy, Task Complete | x
Tests for Destroy & Task Complete include tests for valid and invalid task ids | x
Routes follow RESTful conventions | x
Uses named routes (like `_path`) | x
**Overall** |



Well done on this project, Heather!


Your project hits all the things I was looking for: following Rails best practices, RESTful routing, and CRUD.


The places of improvement that I have noted are:
- You have defined a strong params method `task_params`, but you don't use it anywhere! Your project will be better off using the strong params method
- You have a few unused lines of code


There are also just a few other comments I have


Also, you used some of the live code copy (like "MAKE! THAT! task!"), so don't forget to change small details like that to make this app your own :) :) :) (Not a big deal, just something that might feel nice for yourself)


That being said, your project looks pretty good. Great work!

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
+class TasksController < ApplicationController
+  TASKS = [
+    { name: "cleaning", description: "vacuum" },
+    { name: "dinner", description: "make soup" },
+    { name: "shopping", description: "Go to Target" },
+  ]
```
Comment (tildeee) [inline app/controllers/tasks_controller.rb:6]: Don't forget to refactor your code often, so you remember to delete unused variables like this!

File: app/controllers/tasks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  def new
+    @task = Task.new(name: "Default task")
+  end
+
+  def create
+    # puts params
+    task = Task.new(
+      name: params["task"]["name"],
+      description: params["task"]["description"],
+    )
```
Comment (tildeee) [inline app/controllers/tasks_controller.rb:29]: This is a place where you can refactor this to use `task_params` instead of outlining `name` and `description`

File: app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    if is_successful
+      redirect_to task_path(task.id)
+    else
+      head :not_found
+    end
+  end
+
+  def edit
+    puts params
```
Comment (tildeee) [inline app/controllers/tasks_controller.rb:41]: You'd probably want to delete your `puts`!

File: app/controllers/tasks_controller.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    task_id = params[:id]
+    @task = Task.find_by(id: task_id)
+
+    if @task.nil?
+      redirect_to tasks_path
+    end
+  end
+
+  def update
+    task = Task.find_by(id: params["id"])
```
Comment (tildeee) [inline app/controllers/tasks_controller.rb:51]: Strong params!

File: app/controllers/tasks_controller.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      task.destroy
+      redirect_to tasks_path
+    end
+  end
+
+  def mark_complete
+    task = Task.find_by(id: params[:id])
+    task.toggle(:completed)
+    task.save
+    task.touch
```
Comment (tildeee) [inline app/controllers/tasks_controller.rb:78]: What does this line do? Your tests still pass when I remove this line, and also the app still seems to work

File: app/views/tasks/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+<%= link_to task.name, task_path(task.id), class: "index-name" %>
+
+
+<% if task.completed == false %>
+<%= button_to "Mark Complete", mark_complete_path(task.id), method: :patch, class: "status-button"  %>
+
+<%else%>
+<%= button_to "Mark Incomplete", mark_complete_path(task.id), method: :patch, class: "status-button"   %>
+
+<%end %>
```
Comment (tildeee) [inline app/views/tasks/index.html.erb:21]: this view and the logic in it looks good! You may want to mind your indentation though
